src/ansys/speos/core/generic/constants.py

```diff
@@ -38,3 +42,347 @@
+
+    maximum_nb_of_sequence: int = 10
+    sequence_type: Union[SequenceTypes.by_face, SequenceTypes.by_geometry] = (
+        SequenceTypes.by_geometry
+    )
+
+
+@dataclass
+class GeometryLayerParameters:
+    """Geometry layer parameters."""
+
+    name: str
+    geometry: list
+
+
+@dataclass
+class LayerByFaceParameters:
+    """Layer separation type Parameters  for Face separation."""
+
+    geometries: list[GeometryLayerParameters] = None
+    sca_filtering_types: Union[
+        SCAFilteringTypes.intersected_one_time, SCAFilteringTypes.last_impact
+    ] = SCAFilteringTypes.last_impact
+
+
+@dataclass
+class LayerByIncidenceAngleParameters:
+    """Layer separation type Parameters for Incidence angle separation."""
+
+    incidence_sampling: int = 9
+
+
+class PngBits(str, Enum):
+    """Bit resolution of create PNG image."""
+
+    png_08 = "png_08"
+    png_10 = "png_10"
+    png_12 = "png_12"
+    png_16 = "png_16"
+
+
+class ColorBalanceModeTypes(str, Enum):
+    """Color Balance Mode types without parameters."""
+
+    none = "none"
+    grey_world = "grey_world"
+
+
+@dataclass
+class BalanceModeUserWhiteParameters:
+    """Parameters for Balance mode: User White."""
+
+    red_gain: float = 1.0
+    """Gain value for Red."""
+    green_gain: float = 1.0
+    """Gain value for green."""
+    blue_gain: float = 1.0
+    """Gain value for blue."""
+
+
+@dataclass
+class BalanceModeDisplayPrimariesParameters:
+    """Parameters for Balance mode: Display primaries."""
+
+    red_display_file_uri: Union[str, Path] = ""
+    """Spectrum path of the red display spectrum."""
+    green_display_file_uri: Union[str, Path] = ""
+    """Spectrum path of the green display spectrum."""
+    blue_display_file_uri: Union[str, Path] = ""
+    """Spectrum path of the red display spectrum."""
+
+
+@dataclass
+class ColorParameters:
+    """Color mode Camera Parameter."""
+
+    balance_mode: Union[
+        ColorBalanceModeTypes.none,
+        ColorBalanceModeTypes.grey_world,
+        BalanceModeUserWhiteParameters,
+        BalanceModeDisplayPrimariesParameters,
+    ] = ColorBalanceModeTypes.none
+    """Camera Balance mode."""
+    red_spectrum_file_uri: str = ""
+    """Path to sensitivity spectrum of red Channel."""
+    green_spectrum_file_uri: str = ""
+    """Path to sensitivity spectrum of green Channel."""
+    blue_spectrum_file_uri: str = ""
+    """Path to sensitivity spectrum of blue Channel."""
+
+
+@dataclass
+class MonoChromaticParameters:
+    """Monochromatic Camera Parameters."""
+
+    sensitivity: str = ""
+    """Path to Sensitivity Spectrum."""
+
+
+@dataclass
+class PhotometricCameraParameters:
+    """Photometric Parameters for Camera."""
+
+    color_mode: Union[MonoChromaticParameters, ColorParameters] = field(
+        default_factory=ColorParameters
+    )
+    """Color mode of the Camera Sensor."""
+    layer_type: Union[LayerTypes.none, LayerTypes.by_source] = LayerTypes.none
```

Comment:
```suggestion
    layer_type: Optional[LayerTypes.by_source] = None
```
check post_init below